certification/internal/shell/has_prohibited_packages.go

```diff
@@ -2,33 +2,61 @@ package shell
 
 import (
 	"bufio"
-	"bytes"
-	"os/exec"
+	"strings"
 
 	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/cli"
 	log "github.com/sirupsen/logrus"
 )
 
 type HasNoProhibitedPackagesCheck struct{}
 
 func (p *HasNoProhibitedPackagesCheck) Validate(image string) (bool, error) {
-	stdouterr, err := exec.Command("podman", "run", "-it", "--rm", "--entrypoint", "rpm", image, "-qa", "--queryformat", "%{NAME}\n").CombinedOutput()
+	line, err := p.getDataToValidate(image)
 	if err != nil {
 		log.Error("unable to get a list of all packages in the image")
 		return false, err
 	}
 
-	scanner := bufio.NewScanner(bytes.NewReader(stdouterr))
+	return p.validate(line)
+
+}
+
+func (p *HasNoProhibitedPackagesCheck) getDataToValidate(image string) (string, error) {
+	runOpts := cli.ImageRunOptions{
+		EntryPoint:     "rpm",
+		EntryPointArgs: []string{"-qa", "--queryformat", "%{NAME}\n"},
+		LogLevel:       "debug",
+		Image:          image,
+	}
+	runReport, err := podmanEngine.Run(runOpts)
+	if err != nil {
+		log.Error("unable to get a list of all packages in the image, error: ", err)
+		log.Debugf("Stdout: %s", runReport.Stdout)
+		log.Debugf("Stderr: %s", runReport.Stderr)
+		return "", err
+	}
+	return runReport.Stdout, nil
+}
+
+func (p *HasNoProhibitedPackagesCheck) validate(line string) (bool, error) {
+	scanner := bufio.NewScanner(strings.NewReader(line))
+	prohibitedPackageFound := false
```

Comment:
Giving this a second thought, we can get rid of this variable and use `pkgs` to figure out if the check passes or fails. It would be great to rename `pkgs` to `prohibitedPackages` or anything a bit descriptive.